Dump and merge VTT data to/from external single file

madig opened the pull request:
To support VTT data for instances: https://github.com/daltonmaag/vttLib/issues/19

anthrotype commented on src/vttLib/transfer.py:
```diff
@@ -0,0 +1,72 @@
+    maxp.maxZones = 1
+    maxp.numGlyphs = len(ufo)
+    if ".notdef" not in ufo:
+        maxp.numGlyphs += 1
```
Why +1 if .notdef not in UFO?

madig replied:
Because if increases numGlyphs by 1 when roundtripping (i.e. dumping the data again).

anthrotype commented on src/vttLib/__main__.py:
```diff
@@ -106,28 +95,95 @@ def main(args=None):
+
+
+def vtt_move_ufo_data_to_file(infile, outfile=None, **_):
+    """Write VTT data from a TTX dump to a TTF."""
```
Docstring doesn't match content, copy pasted.

anthrotype commented on src/vttLib/transfer.py:
```diff
@@ -0,0 +1,72 @@
+        font["TSI0"] = fontTools.ttLib.newTable("TSI0")
+    if "TSI2" not in font:
+        font["TSI2"] = fontTools.ttLib.newTable("TSI2")
+    font.importXML(path)
```
If you dump the whole maxp table in the ttx file and then import it as is, it will overwrite other fields that you do not want to change like numGlyphs (a UFO may have additional glyphs or less glyphs since the last dump). That is why we were dumping only a subset of fields to a plist and merging it ourselves without using ttx machinery, which only supports dumping/merging whole tables.

madig commented:
On a second though: don't require ufo2ft and change the test to not care about the diffs introduced by `maxp.recalc()`.

madig committed:
Boilerplate for the coming commits

madig committed:
Add test data

madig committed:
Store VTT data in external dump

This is to support storing hinting data of instances and variable fonts, which are naturally not tied to UFOs.

belluzj commented on src/vttLib/__main__.py:
```diff
@@ -1,103 +1,91 @@
-
-    for subparser in subparsers:
+    parser_dumpfile_from_ufo = parser_group.add_parser(
+        "dumpfile_from_ufo", description="Dump VTT data from UFO3 data to a TTX dump."
```
Please say here that the original data files will be removed, or don't remove the original data files (I would prefer to not remove the input files because it's unexpected for a tool that reads data to also delete it)

belluzj commented:
LGTM

madig committed:
Do not clear out UFO data when dumping to external file
